fv3core/decorators.py

```diff
@@ -335,6 +335,32 @@ def decorator(func) -> FV3StencilObject:
         return decorator(definition)
 
 
+class FixedOriginStencil:
+    """Wrapped GT4Py stencil object explicitly genrating
+    and using the normalized origins."""
+
+    def __init__(self, func, origin, domain, **kwargs):
+        self.normalized_origin = gtscript.gt_definitions.normalize_origin_mapping(
+            origin
+        )
+        self.domain = domain
+        self.func = func
+        self.stencil_object = gtscript.stencil(
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
```

Comment:
so here is where we can add @eddie-c-davis  turning off of synchronize, shouldn't be too hard to do then. He named these objects something else, so checking in with him to coordinate order of operations makes sense.

Reply:
I created a [stencil_wrapper](https://github.com/VulcanClimateModeling/fv3core/blob/ab86522213ddce95692ac06bbf1ed9f504589d95/fv3core/decorators.py#L340) method to encapsulate this behavior.

Reply:
This can be combined